packages/flame/lib/src/sprite_batch.dart

```diff
@@ -315,7 +363,7 @@ class SpriteBatch {
   }) {
     paint ??= _emptyPaint;
 
-    if (!useAtlas) {
+    if (!useAtlas || !_atlasReady) {
```

Comment:
If you switch the order of then and else blocks here, then the condition becomes simpler: `if (useAtlas && _atlasReady)`